pep-0716.rst

```diff
@@ -0,0 +1,319 @@
+  filesystem or to PyPI.
+* Some tools will normalize the project name before emitting them to either
+  ``METADATA`` or to PyPI.
+* Some tools (PyPI) require that the filename and the project name match, without
+  taking normalization into account.
+* Some tools (Artifactory) require that the filenames are not normalized.
+* The above sets of tools do not perfectly overlap in any direction.
+
+We've essentially created a mess where nobody is emitting filenames in quite the
+same way and the normalization rules, first defined in :pep:`503` are being used
+in contexts where it is not appropriate to do so.
+
+
+Rationale
+=========
+
+This PEP follows two guiding principals:
+
+1. Names are provided by people and should be used as is where possible. The
+   name of the project, and how it appears, is a fundamental property of the
+   project.
+2. When interpreting names, tooling should normalize values as much as
+   possible to reduce confusion.
+
+This follows the original intent behind the normalization in :pep:`503`, which
+was designed to be a normalization applied when two computers spoke to each
+other, not as something that would "leak" out into the human-facing areas.
+
+
+Specification
+=============
+
+The project name that is specified by an author ends up flowing through several
+parts of the ecosystem, and each part needs to be considered on its own to determine what
+kind of name (normalized or not) makes sense in that part.
+
+In general, we follow the guiding principals, use the unnormalized name as
+provided by the author wherever possible, and normalize strictly where not.
+
+In some cases, we are simply repeating the status quo, this is done to provide
+clarification and to be explicit which uses were considered as part of this
+PEP.
+
+
+Core Metadata
+-------------
+
+The ``Name`` field **MUST NOT** be normalized when emitting into ``METADATA``
+or ``PKG-INFO``.
+
+The ``Name`` field **MUST NOT** be normalized when uploading to a repository.
+
+The ``Name`` field **SHOULD NOT** be normalized when being presented for display
+to a user.
+
+The ``Name`` field **MUST** be normalized during comparison.
+
+Tools that read the ``Name`` field from a core metadata file **MUST** be prepared
+to accept unnormalized names.
+
+
+pyproject.toml
+--------------
+
+The ``project.name`` key **MUST** be preserved exactly as the author chose to
+represent it, and **MUST** be emitted in this way into ``METADATA`` or
+``PKG-INFO``.
+
+The ``project.name`` field **MUST** be normalized during comparison.
+
+
+.dist-info directories
+----------------------
+
+The directory name follows the pattern of ``{name}-{version}.dist-info``.
+
+The ``name`` field **MUST** be normalized, with any resulting ``-`` escaped to ``_``.
+
+Tools that read an arbitrary ``.dist-info`` directory **MUST** be prepared to
+accept unnormalized values, however tools that work only on *new* ``.dist-info``
+directories **SHOULD** validate that all values are normalized.
```

Comment:
Not grammar, but rather a point of semantics (which may need to be raised on Discourse, but if I don't mention it now, I'll forget 🙂). I'd prefer this to be **MAY**. We should never insist (even mildly) on validation from consumers. Suggesting it is fine.